blueprints/component-test/index.js

```diff
@@ -1,3 +1,65 @@
 module.exports = {
-  description: 'Generates a component unit test.'
+  description: 'Generates a component test.',
+
+  availableOptions: [
+    {
+      name: 'integration',
+      type: Boolean,
+      default: false,
+      aliases: ['i']
+    }
+  ],
+
+  locals: function(options) {
+    var prefix, testOpts, defaultTest;
+    if (options.integration) {
+      prefix = 'Integration: ';
+      testOpts = 'integration: true';
+      defaultTest = "// renders the component on the page\n" +
+             "       this.render('{{" + options.entity.name+ "}}');\n" +
+             "       expect(this.$()).to.be.ok;";
+    } else {
+      prefix = '';
+      testOpts = "// specify the other units that are required for this test\n" +
+             "    // needs: ['component:foo', 'helper:bar']";
+      defaultTest = "// creates the component instance\n" +
+              "      var component = this.subject();\n" +
+              "      expect(component._state).to.equal('preRender');\n\n" +
+              "      // renders the component on the page\n" +
+              "      this.render();\n" +
+              "      expect(component._state).to.equal('inDOM');";
+    }
+
+    return {
+      testPrefix: prefix,
+      testOpts: testOpts,
+      defaultTest: defaultTest,
+      integration: options.integration
+    };
+  },
+
+  fileMapTokens: function(options) {
+    return {
+      __testfolder__: function(options) {
```

Comment:
This introduces a new folder structure for integration tests. This is nice because you can both unit and integration test your components, with different files (either `tests/unit/components/*-test.js` or `tests/integration/components/*-test.js`.